sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -20,12 +29,95 @@
 from ...model_selection import train_test_split
 from ...preprocessing import LabelEncoder
 from ._gradient_boosting import _update_raw_predictions
-from .common import Y_DTYPE, X_DTYPE, X_BINNED_DTYPE
+from .common import Y_DTYPE, X_DTYPE, X_BINNED_DTYPE, G_H_DTYPE
 
 from .binning import _BinMapper
 from .grower import TreeGrower
-from .loss import _LOSSES
-from .loss import BaseLoss
+
+
+_LOSSES = _LOSSES.copy()
+# TODO: Remove least_squares and least_absolute_deviation in v1.2
+_LOSSES.update(
+    {
+        "least_squares": HalfSquaredError,
+        "least_absolute_deviation": AbsoluteError,
+        "poisson": HalfPoissonLoss,
+        "binary_crossentropy": HalfBinomialLoss,
+        "categorical_crossentropy": HalfMultinomialLoss,
+    }
+)
+
+
+def _init_gradients_and_hessians(constant_hessian, n_samples, prediction_dim):
```

Comment:
Should this function be a part of the Loss API as a method? The Loss objects know how to write to gradients and hessians. I think it makes sense they would know how to construct them.

Reply:
Good idea.

Reply:
What I like in the original HGBT implementation is that raw_prediction, gradients and hessians were always 2d arrays of shape `(n_trees_per_iteration_, n_samples)`. 
The new loss classes have the axes reversed to shape `(n_samples, ...)`, which is similar to the shape of `X`. On top, they mostly prefer 1d arrays, which feels more natural to me for regression on a single target and for binary classification (which is regression on a single target :wink:).